src/test/java/io/jenkins/plugins/sprp/EndToEndTest.java

```diff
@@ -0,0 +1,119 @@
+                "    - name: Tests\n" +
+                "      steps:\n" +
+                "        - stepName: sh\n" +
+                "          defaultParameter: ./scripts/hello\n" +
+                "\n" +
+                "archiveArtifacts:\n" +
+                "    - Jenkinsfile.yaml\n" +
+                "    - scripts/hello.sh\n" +
+                "\n";
+
+        String pipelineScriptLinux = "pipeline {\n" +
+                "\tagent any\n" +
+                "\tstages {\n" +
+                "\t\tstage('First') {\n" +
+                "\t\t\tsteps {\n" +
+                "\t\t\t\tsh './scripts/hello.sh'\n" +
+                "\t\t\t\tsleep 5\n" +
+                "\t\t\t}\n" +
+                "\t\t}\n" +
+                "\t\tstage('Build') {\n" +
+                "\t\t\tsteps {\n" +
+                "\t\t\t\tsh './scripts/build.sh'\n" +
+                "\t\t\t}\n" +
+                "\t\t\tpost {\n" +
+                "\t\t\t\tsuccess {\n" +
+                "\t\t\t\t\tarchiveArtifacts artifacts: '**/target/*.jar'\n" +
+                "\t\t\t\t\tarchiveArtifacts artifacts: 'Jenkinsfile.yaml'\n" +
+                "\t\t\t\t\tarchiveArtifacts artifacts: 'scripts/hello.sh'\n" +
+                "\t\t\t\t}\n" +
+                "\t\t\t}\n" +
+                "\t\t}\n" +
+                "\t\tstage('Tests') {\n" +
+                "\t\t\tsteps {\n" +
+                "\t\t\t\tsh './scripts/hello.sh'\n" +
+                "\t\t\t}\n" +
+                "\t\t}\n" +
+                "\t}\n" +
+                "}\n";
+
+        String pipelineScriptWindows = "pipeline {\n" +
+                "\tagent any\n" +
+                "\tstages {\n" +
+                "\t\tstage('First') {\n" +
+                "\t\t\tsteps {\n" +
+                "\t\t\t\tsh './scripts/hello.bat'\n" +
+                "\t\t\t\tsleep 5\n" +
+                "\t\t\t}\n" +
+                "\t\t}\n" +
+                "\t\tstage('Build') {\n" +
+                "\t\t\tsteps {\n" +
+                "\t\t\t\tsh './scripts/build.bat'\n" +
+                "\t\t\t}\n" +
+                "\t\t\tpost {\n" +
+                "\t\t\t\tsuccess {\n" +
+                "\t\t\t\t\tarchiveArtifacts artifacts: '**/target/*.jar'\n" +
+                "\t\t\t\t\tarchiveArtifacts artifacts: 'Jenkinsfile.yaml'\n" +
+                "\t\t\t\t\tarchiveArtifacts artifacts: 'scripts/hello.sh'\n" +
+                "\t\t\t\t}\n" +
+                "\t\t\t}\n" +
+                "\t\t}\n" +
+                "\t\tstage('Tests') {\n" +
+                "\t\t\tsteps {\n" +
+                "\t\t\t\tsh './scripts/hello.bat'\n" +
+                "\t\t\t}\n" +
+                "\t\t}\n" +
+                "\t}\n" +
+                "}\n";
+
+        StringInputStream yamlInputStream = new StringInputStream(yaml);
+        if(jenkinsRule.createLocalLauncher().isUnix()){
+            assertEquals(pipelineScriptLinux,
+                    new YamlToPipeline().generatePipeline(yamlInputStream,
```

Comment:
I would suggest maybe executing the `generatePipeline` function to get the actual  results before the if statement to remove some code duplication.